packages/dd-trace/src/appsec/reporter.js

```diff
@@ -225,6 +242,53 @@ function reportWafInit (wafVersion, rulesVersion, diagnosticsRules = {}, success
   incrementWafInitMetric(wafVersion, rulesVersion, success)
 }
 
+function reportSuccessfulWafUpdate (product, rcConfigId, diagnostics) {
+  for (const configKey of WAF_DIAGNOSTICS_CONFIG_KEYS_TO_REPORT) {
+    if (!diagnostics[configKey]) continue
+
+    if (diagnostics[configKey].error) {
+      telemetryLogCh.publish({
+        message: diagnostics[configKey].error,
+        level: 'ERROR',
+        tags: {
+          log_type: `rc::${product.toLowerCase()}::diagnostic`,
+          appsec_config_key: configKey,
+          rc_config_id: rcConfigId
+        }
+      })
```

Comment:
I would abstract this in a small helper with the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))

Reply:
New helper has been implemented to publish these diagnostics log messages.

Reply:
Thanks, the abstraction is easier to read for me and I find it much cleaner :)